### PR TITLE
Add the currency sign to IOU request flow

### PR DESCRIPTION
**Pull request by iwiznia**
### Fixed Issues

Fixes https://github.com/Expensify/Expensify/issues/164533

### Tests
Request Money

- Click the + icon and select Request Money
- Enter $50 on the IOUAmount page and click next
- Observe that the header of the IOUParticipants page reads Request $50
image
- Select a participant to move on to the IOUConfirm page
- Observe that the header & confirmation button both read Request $50
image

Split Bill

- Click the + icon and select Split Bill
- Enter $50 on the IOUAmount page and click next
- Observe that the header of the IOUParticipants page reads Split $50
image
- Select a couple of participants and click next
- Observe that the header of the IOUConfirm page also reads Split $50
image

### Tested On

- [x] Web
- [ ] Mobile Web
- [ ] Desktop
- [ ] iOS
- [ ] Android

### CODE REVIEW

**Comment by Julesssss** [comment]
@iwiznia FYI conflicts

File: src/components/IOUConfirmationList.js [hunk 1]
```diff
@@ -224,6 +224,14 @@ class IOUConfirmationList extends Component {
...
 
     render() {
+        const buttonText = this.props.translate(
+            this.props.hasMultipleParticipants ? 'common.split' : 'iou.request', {
+                amount: this.props.numberFormat(
+                    this.props.iouAmount,
+                    {style: 'currency', currency: this.props.selectedCurrency},
+                ),
+            },
+        );
```
Comment (Julesssss) [inline src/components/IOUConfirmationList.js:234]: We've got a condition, translation, and number format all in one chunk. Maybe separating out `amount` would simplify this statement a bit.

File: src/languages/en.js [hunk 2]
```diff
@@ -8,7 +8,7 @@ export default {
         attachment: 'Attachment',
         to: 'To',
         optional: 'Optional',
-        split: 'Split',
+        split: ({amount}) => `Split ${amount}`,
```
Comment (Julesssss) [inline src/languages/en.js:11]: I think it shouldn't be common anymore, as it must have an amount. It should be listed beneath `iou.request` IMO

File: src/pages/iou/IOUModal.js [hunk 3]
```diff
@@ -126,9 +126,16 @@ class IOUModal extends Component {
...
-                : this.props.translate('iou.request', {amount: this.state.amount})}`;
+            return this.props.translate(
+                this.props.hasMultipleParticipants ? 'common.split' : 'iou.request', {
+                    amount: this.props.numberFormat(
+                        this.state.amount, {
+                            style: 'currency',
+                            currency: this.state.selectedCurrency,
+                        },
+                    ),
+                },
```
Comment (Julesssss) [inline src/pages/iou/IOUModal.js:137]: Same here. Separating this into two steps would be a lot simpler.

**Review by Julesssss — CHANGES_REQUESTED** [review]
Requested a few changes

File: src/languages/en.js (hunk 2 above)
Comment (iwiznia) [reply src/languages/en.js:11]: The fact that it has an amount does not make a difference. We are using this in 3 places, so it is pretty common to me.

**Comment by iwiznia** [comment]
Updated

File: src/languages/en.js (hunk 2 above)
Comment (Beamanator) [reply src/languages/en.js:11]: I see `iou.request` used in the same places as `common.split` - both in `IOUConfirmationList.js` and `IOUModal.js`.

My opinion is that adding an `${amount}` in the text makes it very `iou`-related so I'd agree with Jules.

File: src/components/IOUConfirmationList.js (hunk 1 above)
Comment (Beamanator) [reply src/components/IOUConfirmationList.js:234]: Looking at it like this, I'm not sure it offers too much of an improvement 🤷 

Screen Shot 2021-05-19 at 10 27 49 PM

**Comment by iwiznia** [comment]
Updated

**Review by Beamanator — APPROVED** [review]
Tested and works well on Desktop & iOS 👍

**Review by Julesssss — APPROVED** [review]

**Pull request merged**